### PR TITLE
HDDS-8182. Add volume and container choosing policy

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

This ticket is to add ContainerChoosingPolicy and VolumeChoosingPolicy for DiskBalancer Service.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8182

## How was this patch tested?

unit test.

### CODE REVIEW

**Comment by symious** [comment]
@ChenSammi Could you help to review the PR?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/DiskBalancerService.java [hunk 1]
```diff
@@ -110,6 +119,19 @@ public DiskBalancerService(OzoneContainer ozoneContainer,
...
+      volumeChoosingPolicy = conf.getClass(
+          HDDS_DATANODE_DISK_BALANCER_VOLUME_CHOOSING_POLICY,
+          DefaultVolumeChoosingPolicy.class,
+          VolumeChoosingPolicy.class).newInstance();
+      containerChoosingPolicy = conf.getClass(
+          HDDS_DATANODE_DISK_BALANCER_CONTAINER_CHOOSING_POLICY,
+          DefaultContainerChoosingPolicy.class,
+          ContainerChoosingPolicy.class).newInstance();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/DiskBalancerService.java:132]: Please log error meg here.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultContainerChoosingPolicy.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * Choose a container from specified volume, make sure it's not being balancing.
+ */
+public class DefaultContainerChoosingPolicy implements ContainerChoosingPolicy {
+  public static final Logger LOG = LoggerFactory.getLogger(
+      DefaultContainerChoosingPolicy.class);
+
+  @Override
+  public ContainerData chooseContainer(OzoneContainer ozoneContainer,
+      HddsVolume hddsVolume, Set<Long> inProgressContainerIDs) {
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultContainerChoosingPolicy.java:40]: I think we should turn this API into a batch API. 
ozoneContainer.getController().getContainers(hddsVolume) is a quite expensive operation. Try to mostly leverage the result in one chooseContainer call.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultVolumeChoosingPolicy.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            Double.compare(
+                (double) (v2.getUsedSpace() + deltaMap.getOrDefault(v2, 0L)) /
+                    v2.getCapacity(),
+                (double) (v1.getUsedSpace() + deltaMap.getOrDefault(v1, 0L)) /
+                    v1.getCapacity()))
+        .collect(Collectors.toList());
+
+    // Can not generate DiskBalancerTask if we have less than 2 results
+    if (volumes.size() <= 1) {
+      return null;
```
Comment (ChenSammi) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultVolumeChoosingPolicy.java:61]: Please log a debug level msg for this case.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerStatus.java [hunk 4]
```diff
@@ -45,4 +45,4 @@ public boolean isRunning() {
   public DiskBalancerConfiguration getDiskBalancerConfiguration() {
     return diskBalancerConfiguration;
   }
-}
\ No newline at end of file
+}
```
Comment (ChenSammi) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerStatus.java:48]: Please revert this unnecessary change in this and the following two files.

Comment (symious) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerStatus.java:48]: The checkstyle will prompt the following warnings:
```
hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerStatus.java
 1: File does not end with a newline.
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultContainerChoosingPolicy.java (hunk 2 above)
Comment (symious) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/policy/DefaultContainerChoosingPolicy.java]: Added a new parameter "targetSize" for batch retrieve of containers here.

**Comment by ChenSammi** [comment]
LGTM,  +1.
